### PR TITLE
[package.json]: Upgrade `ethereumjs-wallet` from `0.6.3` to `^1.0.1`

### PR DESCRIPTION
**Pull request by PetarKirov**
Since version `1.0.0` [ethereumjs-wallet][1] [uses `scrypt-js@^3.0.1`][2]
instead of `scrypt.js@^0.3.0` or `@web3-js/scrypt-shim@^0.1.0`. These
older packages (with `@web3-js/scrypt-shim` currently deprecated), used
the native dependency `scrypt` which which fails to compile with
Node.js 14.

Even though `scrypt` is now marked as optional, this looses unnecessary
amount of time for everyone involved who's using Node.js 14.

`scrypt-js` is a pure JS implementation, and as such is fully compatible
with newer Node.js versions.

Also delete the whole `typings/ethereumjs-wallet/` folder as
`ethereumjs-wallet` is written in TypeScript and there's no need for us
to duplicate the typings (wrongly).

[1]: https://github.com/ethereumjs/ethereumjs-wallet
[2]: https://github.com/ethereumjs/ethereumjs-wallet/commit/71d1c44d08948e4226b9ea75324e9e5059f008df

### CODE REVIEW

**Comment by eggplantzzz** [comment]
Thanks for the PR @PetarKirov! We'll get this reviewed, are you able to resolve the failure in CI?

File: packages/core/lib/mnemonics/mnemonic.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const Config = require("@truffle/config");
 const defaultUserConfig = Config.getUserConfig();
 const bip39 = require("ethereum-cryptography/bip39");
 const { wordlist } = require("ethereum-cryptography/bip39/wordlists/english");
-const hdkey = require("ethereumjs-wallet/hdkey");
+const hdkey = require("ethereumjs-wallet/dist/hdkey");
```
Comment (eggplantzzz) [inline packages/core/lib/mnemonics/mnemonic.js:14]: I don't think you should try to go into the `dist` folder this way. I think you should do
```
const { hdkey } = require("ethereumjs-wallet");
```
Check out their example on the repo https://github.com/ethereumjs/ethereumjs-wallet#hd-wallet-api

File: packages/hdwallet-provider/src/index.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import * as bip39 from "ethereum-cryptography/bip39";
 import { wordlist } from "ethereum-cryptography/bip39/wordlists/english";
 import * as EthUtil from "ethereumjs-util";
 import ethJSWallet from "ethereumjs-wallet";
-import EthereumHDKey from "ethereumjs-wallet/hdkey";
+import EthereumHDKey from "ethereumjs-wallet/dist/hdkey";
```
Comment (eggplantzzz) [inline packages/hdwallet-provider/src/index.ts:6]: Same comment as above.

**Comment by PetarKirov** [comment]
Thanks for the review @eggplantzzz, I'll address your comments later today and I'll try to fix the CI issues.

File: packages/core/lib/mnemonics/mnemonic.js (hunk 1 above)
Comment (PetarKirov) [reply packages/core/lib/mnemonics/mnemonic.js]: You're right - all fixed now!

**Comment by PetarKirov** [comment]
@eggplantzzz I addressed your comments and the CI is now green.

**Comment by PetarKirov** [comment]
@gnidan @haltman-at Hey guys, could you please a look at my pull request?

**Review by haltman-at — APPROVED** [review]
Looks good to me, thanks!

**Review by gnidan — APPROVED** [review]
And to me!

**Comment by PetarKirov** [comment]
Thanks for the swift review!

**Comment by PetarKirov** [comment]
@gnidan @haltman-at can you guys also push a new release to [npm](https://www.npmjs.com/package/@truffle/hdwallet-provider), pls?

**Comment by eggplantzzz** [comment]
@PetarKirov We usually release on Thursdays or Fridays so you should have one out by then!

**Comment by PetarKirov** [comment]
@eggplantzzz sounds like a good Christmas present, thanks :)

**Comment by gnidan** [comment]
I'll be handling the release this week; expect it on the 26th!

**Pull request merged**